oshi-core/src/main/java/oshi/hardware/platform/unix/openbsd/OpenBsdCentralProcessor.java

```diff
@@ -241,14 +244,33 @@ protected long queryInterrupts() {
      *            A buffer containing a long array
      * @return The long array
      */
-    private static long[] cpTimeToTicks(Memory m) {
+    private static NativeLong[] cpTimeToTicks(Memory m) {
         if (m != null) {
             if (m.size() == 5 * Native.LONG_SIZE) {
                 return new CpTime(m).cpu_ticks;
             } else if (m.size() == 6 * Native.LONG_SIZE) {
                 return new CpTimeNew(m).cpu_ticks;
             }
         }
+        return new NativeLong[0];
+    }
+
+    /**
+     * Parse memory buffer returned from sysctl kern.cptime2 to array of 5 or 6
+     * longs depending on version
+     *
+     * @param m
+     *            A buffer containing a long array
+     * @return The long array
+     */
+    private static long[] cpTime2ToTicks(Memory m) {
```

Comment:
it's a private method so not to important, but since it's parsing `kern.cp_time` you may want to change this method name

Reply:
Good point.   There's a bit of inconsistency in the source.
 - `kern.cptime` name points to `KERN_CPTIME` constant points to internal code with a `cp_time` array of native-sized `long`
 - `kern.cptimes` name ponits to `KERN_CPTIME2` constant points to internal BSD `sysctl_cptime2()` pointing to internal code with a structure containing `spc_cp_time` array of 64-bit `long`.

I initially used the structure approach to make array access easy but that's not really needed and we now have 4 pseudo-structures. Let me just rewrite it to return arrays directly from the memory buffer.   Still need a separate method for the 32/64-bit vs. 64-bit-only approaches.